application/core/LSUserIdentity.php

```diff
@@ -85,6 +79,16 @@ public function authenticate()
             }
         }
 
+        /* Check user exist, valid and not expired after plugin actions */
+        if ($result->isValid()) {
+            /** @var \User|null */
+            $user = User::model()->notexpired()->active()->findByAttributes(array('users_name' => $this->username));
```

Comment:
The intention here is to check if the user is ok to login.
Instead of chainning the rules thrugh scopes, wouldn't it be better to:
- Look for the user
- Create a method canLogin()
- Call that method.

That way the method can easily be enriched with more rules.

Reply:
:+1: